rfcs/convergence/slot-null-rendering.md

```diff
@@ -0,0 +1,261 @@
+
+### Option 1: Make `shorthands` optional
+
+This solves this problem by verifying if the shorthand is `undefined` or not.
+By verifying if shorthand is `undefined` we can opt for null rendering without compromising the cases which native slots don't have children
+
+- `getSlots` should stop verifying null rendering by the existence of the property `children`, instead depending on the existence of the shorthand itself
+
+```tsx
+// react-utilities/src/compose/getSlots.ts
+function getSlots(state, slotNames) {
+  // ...
+  if (state[name] === undefined) {
+    slots[name] = nullRender;
+    continue;
+  }
+  // ...
+}
+```
+
+- Since some slots may have default props, verification must be done on shorthand declaration to ensure that those slots will be null rendered even presenting default props
+
+```diff
+export function resolveShorthand(
+  value,
+  defaultProps,
++  {optional = true}
++) {
++  // verify if shorthand is undefined
++  if (value === null || (value === undefined && optional)) {
++    return undefined;
++  }
+}
+```
+
+The problem with this approach is that the shorthand signature now will carry with it an `undefined` value, making overrides a little less elegant:
+
+```ts
+export const CustomAccordionHeader = React.forwardRef<HTMLElement, AccordionHeaderProps>((props, ref) => {
+  const state = useAccordionHeader(props, ref);
+
+  if (state.icon) {
+    state.icon.onClick = () => {
+      /**
+       * this is an override,
+       * if this was a hook than it would need
+       * to be declared before the conditional
+       **/
+    };
+  }
+
+  useAccordionHeaderStyles(state);
+  return renderAccordionHeader(state);
+});
+```
+
+In cases where we simply want to ensure that the slot should be rendered even when `shorthand` is `undefined` than `{optional: false}` should be provided
+
+```ts
+export const useState = (props, ref) => {
+  return {
+    ...props,
+    slot1: resolveShorthand(props.slot1), // undefined
+    slot2: resolveShorthand(props.slot2, {}, { optional: false }), // {}
+  };
+};
+```
+
+### Option 2: Symbol for null rendering
+
+This solves this problem by verifying if the shorthand has a special symbol or not.
+By verifying this symbol we can opt for null rendering without compromising the cases which native slots don't have children.
+This is very similar to Option 1 approach but without the downside of having undefined as part of `ObjectShorthandProps` interface.
+
+- `getSlots` should stop verifying null rendering by the existence of the property `children`, instead depending on the existence of the symbol
+
+```tsx
+// react-utilities/src/compose/getSlots.ts
+function getSlots(state, slotNames) {
+  // ...
+  if (typedState[name][nullRenderSymbol]) {
+    slots[name] = nullRender;
+    continue;
+  }
+  // ...
+}
+```
+
+- Since some slots may have default props, verification must be done on shorthand declaration to ensure that those slots will be null rendered even presenting default props
```

Comment:
nit:
```suggestion
- Since some slots may have default props, verification must be done on shorthand declaration to ensure that those slots will be null rendered even when presented with default props
```